JumpscaleCore/clients/test/test_sonic_client.py

```
@@ -0,0 +1,318 @@
+            self.assertTrue(self.client.push(self.COLLECTION, self.BUCKET, article_id, content))
+
+        self.info("Check the count of indexed search data in the collection and bucket")
+        self.assertEquals(self.client.count(self.COLLECTION, self.BUCKET), 6)
+
+        self.info("Push the data to sonic server, with different collection and bucket name")
+        collection_new = "COLL_{}".format(randint(1, 1000))
+        bucket_new = "BUCKET_{}".format(randint(1, 1000))
+        self.client.push(collection_new, bucket_new, "post_1", "test push with different data")
+
+        self.info("Check the count of indexed search data in the new collection and bucket")
+        self.assertEquals(self.client.count(collection_new, bucket_new), 2)
+
+    def test004_query_with_exists_collection_and_bucket(self):
+        """
+        TC 526
+        Test Case to test query with an exists collection and bucket name.
+
+        **Test scenario**
+        #. Push the data to sonic server.
+        #. Query to those data, and check the output.
+        """
+
+        self.info("Push the data to sonic server")
+        for article_id, content in self.data.items():
+            self.assertTrue(self.client.push(self.COLLECTION, self.BUCKET, article_id, content))
+
+        self.assertEqual(len(self.client.query(self.COLLECTION, self.BUCKET, "love")), 2)
+        self.assertEqual(
+            sorted(self.client.query(self.COLLECTION, self.BUCKET, "love")), sorted(['post:5', 'post:4'])
+        )
+
+    def test005_query_with_non_exists_collection_and_bucket(self):
+        """
+        TC 527
+        Test Case to test query with non exists collection and bucket.
+
+        **Test scenario**
+        #. Query for non exists collection and bucket.
+        """
+
+        self.info("Query for non exists collection and bucket.")
+        self.assertEqual(len(self.client.query("RANDOM_COLLECTION", "RANDOM_BUCKET", "test")), 0)
```

This should throw an exception, please open a ticket